zap/src/main/java/org/parosproxy/paros/Constant.java

```diff
@@ -1271,6 +1276,55 @@ private static void upgradeFrom2_11_1(XMLConfiguration config) {
         config.setProperty("hud.enableTelemetry", null);
     }
 
+    static void upgradeFrom2_14_0(XMLConfiguration config) throws ConfigurationException {
+        List<HierarchicalConfiguration> existingTagScanners =
+                config.configurationsAt("pscans.autoTagScanners.scanner");
+        List<Object> existingDescs = config.getList("pscans.autoTagScanners.scanner.name");
+
+        String leadingAutoTagScannersKey = "pscans.autoTagScanners.scanner(";
+        String trailingNameKey = ").name";
+        String trailingTypeKey = ").type";
+        String trailingConfigKey = ").config";
+        String trailingResHeadRegex = ").resHeadRegex";
+        String trailingEnableKey = ").enabled";
+
+        int index = existingTagScanners.size(); // Size is next index due to 0th start
+
+        if (!existingDescs.contains("json_extended")) {
+            config.addProperty(
+                    leadingAutoTagScannersKey + index + trailingNameKey, "json_extended");
+            config.addProperty(leadingAutoTagScannersKey + index + trailingTypeKey, "TAG");
+            config.addProperty(leadingAutoTagScannersKey + index + trailingConfigKey, "JSON");
+            config.addProperty(
+                    leadingAutoTagScannersKey + index + trailingResHeadRegex,
+                    "content-type:\\s{0,10}.{1,20}\\/.{0,100}json");
+            config.addProperty(leadingAutoTagScannersKey + index + trailingEnableKey, false);
+        }
+
+        if (!existingDescs.contains("response_yaml")) {
+            ++index;
```

Comment:
These should be moved to the previous if as the index might not need to be incremented always.